libraries/src/CMS/Dispatcher/Dispatcher.php

```
@@ -62,13 +70,26 @@
 	 */
 	public function __construct(CMSApplication $app, \JInput $input = null)
 	{
-		if (empty($this->namespace))
+		$this->app   = $app;
+		$this->input = $input ?: $app->input;
+
+		// If option is not provided, detect it from dispatcher class name, ie ContentDispatcher
+		if (empty($this->option))
 		{
-			throw new \RuntimeException('Namespace can not be empty!');
```

I would throw an exception when the namespace is empty.